source/docs/software/basic-programming/dimensional-analysis.rst

```diff
@@ -0,0 +1,88 @@
+
+Handing Units in Code
+---------------------
+
+It can be tricky to handle units consistently in code unless we help ourselves with code standards and/or tooling.  Here are some helpful standards for keeping units straight, regardless of programming language:
+
+ * Use telemetry and logs.
+    - The easiest way to track down a conversion error is to find where in code the values stop making physical sense.  This means we have to log all of our relevant code values.  WPILib includes :ref:`telemetry tooling <docs/software/telemetry/telemetry:Adding Telemetry to Robot Code>` to help record and display robot state.
+ * Break up long expressions with complicated units.
+    - Some unit conversion factors used in robot code (especially those concerning translating from motor rotations to real-world distances) can include many parts and be pretty ugly.  Always try to define the parts of these expressions separately, building up the final conversion ratio from the individually-defined parts.
+ * *Never* blindly assume you can compensate with an empirical "fudge factor".
+    - Sometimes, it can be tempting to bypass a known-but-lengthy chain of unit conversions by taking an "empirical shortcut".  For example, we might be tempted to get the conversion ratio from drive motor rotations to wheel distance by simply running the drive, measuring the displacement and the motor rotation, and dividing the two.  This can work, but only in some cases and at the cost of code clarity and maintainability, since these "shortcut" conversion factors are dimensionally-complex (see previous bullet-point) and externally inscrutable.  In some cases this methodology will not work at all, because the mathematical relationship be more complicated than imagined.
+    - It is still appropriate to use more-accurate empirically-measured conversion ratios *in place of* approximate theoretical ones.  For example, wheel diameter as reported on product sheets is usually nominal and does not reflect the rolling diameter of the wheel on FRC carpet to great accuracy.  To compensate for this, it is a good idea to use an empirically-measured value for the wheel diameter, calculated from a test routine like the one described above.  This is fine as long as the units of the empirically-measured factor are simple, and there is a known theoretical value that it is replacing.
+
+Handling Units in C++
+^^^^^^^^^^^^^^^^^^^^^
+
+For details on managing units and dimensions in C++, see :ref:`docs/software/basic-programming/cpp-units:The C++ Units Library`.
+
+Handling Units in Java
+^^^^^^^^^^^^^^^^^^^^^^
+
+Because Java's type system is not really capable of enforcing dimensional correctness at compile-time without major runtime cost and syntax bloat, there's no units library included as part of WPILibJ.  This makes managing units in Java somewhat difficult, and even experienced developers will sometimes make mistakes.
```

Comment:
```suggestion
Because Java's type system is not really capable of enforcing dimensional correctness at compile-time without major runtime cost and syntax bloat, there's no units library included as part of WPILib.  This makes managing units in Java somewhat difficult, and even experienced developers will sometimes make mistakes.
```

We already know this is specific for Java and we don't really mention WPILibJ anywhere else on the site.

Reply:
This section is OBE now with https://github.com/wpilibsuite/allwpilib/pull/5371